Models/ProtoModel.cpp

```diff
@@ -128,6 +135,10 @@ QVariant ProtoModel::data(int row, int column) const {
 }
 
 QVariant ProtoModel::data(const QModelIndex &index, int role) const {
+  if (!index.isValid()) {
+    qWarning() << index << "provided to ProtoModel::data for role" << role << "is invalid.";

```

Comment:
I love how every strange, far-fetched thing I tell you to do, Qt has a builtin for.

Reply:
Actually that's kind of funny now that you mention it. Kind of a bit of a bitter sweet irony I'd say.

Reply:
No, it just means that Qt is keeping up with the trends. This sort of logger is becoming prevalent, and it's nice to see it included in the framework. It makes a ton of sense for the UI framework to just handle this, and it saves us a bunch of code. All while still giving us access to the information I wanted (eg, source file + line number). It's great.

Reply:
Sure, I get that, I just thought you were being tongue in cheek too. But ok, great!